src/cpu/aarch64/matmul/brgemm_matmul_utils.cpp

```diff
@@ -0,0 +1,1110 @@
+            || bgmmc.src_tag == adbc);
+    // For batched problems with plain A and C and fully broadcasted across B
+    // we can merge all the batch dimensions into M if broadcast strategies
+    // set is limited for binary post-ops
+    const bool plain_A_layout = bm_conf_utils.check_is_plain(bgmmc.src_tag)
+            || treat_transposed_A_as_plain;
+    const bool merge_batch_dims_into_M = bgmmc.batch > 1
+            && bgmmc.bcast_B_desc.bcast_across_all_batch_dims
+            && bm_conf_utils.check_is_plain(bgmmc.dst_tag) && plain_A_layout
+            && post_ops_ok(
+                    bgmmc, attr, dst_d, true /* limit_bcast_strategies_set */);
+    if (merge_batch_dims_into_M) {
+        bgmmc.M *= bgmmc.batch;
+        bgmmc.batch = 1;
+    }
+
+    // runtime A stride wrt M dimension is not acceptable
+    if (is_runtime_value(helper.get_a_stride(bgmmc.ndims - 2)))
+        return status::unimplemented;
+
+    // runtime A stride wrt K dimension is acceptable for transpose A and
+    // runtime M case only
+    const bool stride_A_wrt_K_dim_ok = IMPLICATION(
+            is_runtime_value(helper.get_a_stride(bgmmc.ndims - 1)),
+            bgmmc.transposed_A && bgmmc.is_runtime_M);
+    if (!stride_A_wrt_K_dim_ok) return status::unimplemented;
+
+    // runtime A strides wrt batch dimensions are acceptable for runtime M case
+    // only
+    for (int b = 0; b < bgmmc.batch_ndims; b++) {
+        if (!IMPLICATION(is_runtime_value(helper.get_a_stride(b)),
+                    bgmmc.is_runtime_M))
+            return status::unimplemented;
+    }
+
+    const bool lda_is_big_2pow = false;
+    const bool is_copy_a_required
+            = bgmmc.wei_zp_type != brgemm_broadcast_t::none
+            || bgmmc.transposed_A || lda_is_big_2pow;
+    bgmmc.use_buffer_a = is_copy_a_required;
+
+    // Supported computation with copy only part of A related to K_tail if
+    // is_copy_a_required == true, but the current performance measurements
+    // show worse performance for it in comparison with copy whole A approach
+    // (especially for big K sizes).
+    bgmmc.use_buffer_a_tail_only = false;
+
+    const int dmax = nstl::min(bgmmc.ndims, 3);
+    for (int d = 0; d < dmax; d++) {
+        int dim = bgmmc.ndims - 1 - d;
+        bgmmc.A_strides[d] = bgmmc.a_dt_sz * src_d.blocking_desc().strides[dim];
+        bgmmc.B_strides[d]
+                = bgmmc.b_dt_sz * weights_d.blocking_desc().strides[dim];
+        bgmmc.C_strides[d] = bgmmc.c_dt_sz * dst_d.blocking_desc().strides[dim];
+    }
+
+    // We need to correct A_strides if batched dimensions are merged in M and
+    // A layout is formally transposed but could be treated as plain
+    if (merge_batch_dims_into_M && treat_transposed_A_as_plain) {
+        bgmmc.A_strides[1] = bgmmc.A_strides[2];
+    }
+
+    // BF32 'Hint' Heuristic:
+    // Under the following conditions, F32 through AVX512_CORE performs better
```

Comment:
(random spot): Please remove avx512 and avx2:
- If this code works for sve sounterparts please rename it
- if this is a dead code please remove it

Reply:
@igorsafo, we have removed avx keywords, kindly check the latest changes.